crypto/src/main/java/org/web3j/crypto/MnemonicUtils.java

```diff
@@ -0,0 +1,179 @@
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Provides utility methods to generate random mnemonics and also generate
+ * seeds from mnemonics.
+ *
+ * @see <a href="https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki">Mnemonic code
+ * for generating deterministic keys</a>
+ *
+ * @author Ali Dehghani
+ */
+public class MnemonicUtils {
+
+    private static final int SEED_ITERATIONS = 2048;
+    private static final int SEED_KEY_SIZE = 512;
+    private static final List<String> WORD_LIST = populateWordList();
+
+    /**
+     * The mnemonic must encode entropy in a multiple of 32 bits. With more entropy security is
+     * improved but the sentence length increases. We refer to the initial entropy length as ENT.
+     * The allowed size of ENT is 128-256 bits.
+     *
+     * <h3>Mnemonic generation algorithm</h3>
+     * Given a randomly generated initial entropy of size ENT, first a checksum is generated by
+     * taking the first {@code ENT / 32} bits of its SHA256 hash. This checksum is appended to
+     * the end of the initial entropy. Next, these concatenated bits are split into groups of
+     * 11 bits, each encoding a number from 0-2047, serving as an index into a wordlist. Finally,
+     * we convert these numbers into words and use the joined words as a mnemonic sentence.
+     *
+     * @param initialEntropy The initial entropy to generate mnemonic from
+     * @return The generated mnemonic
+     * @throws IllegalArgumentException If the given entropy is invalid
+     * @throws NoSuchAlgorithmException If we couldn't find a SHA-256 provider
+     */
+    public static String generateMnemonic(byte[] initialEntropy) throws NoSuchAlgorithmException {
+        validateInitialEntropy(initialEntropy);
+
+        int ent = initialEntropy.length * 8;
+        int checksumLength = ent / 32;
+
+        byte checksum = calculateChecksum(initialEntropy);
+        boolean[] bits = convertToBits(initialEntropy, checksum);
+
+        int iterations = (ent + checksumLength) / 11;
+        StringBuilder mnemonicBuilder = new StringBuilder();
+        for (int i = 0; i < iterations; i++) {
+            int index = toInt(nextElevenBits(bits, i));
+            mnemonicBuilder.append(WORD_LIST.get(index));
+
+            boolean notLastIteration = i < iterations - 1;
+            if (notLastIteration) {
+                mnemonicBuilder.append(" ");
+            }
+        }
+
+        return mnemonicBuilder.toString();
+    }
+
+    /**
+     * To create a binary seed from the mnemonic, we use the PBKDF2 function with a
+     * mnemonic sentence (in UTF-8 NFKD) used as the password and the string "mnemonic"
+     * + passphrase (again in UTF-8 NFKD) used as the salt. The iteration count is set
+     * to 2048 and HMAC-SHA512 is used as the pseudo-random function. The length of the
+     * derived key is 512 bits (= 64 bytes).
+     *
+     * @param mnemonic The input mnemonic
```

Comment:
I think it's worth expanding slightly on the mnemonic format in the parameter description.

Reply:
Done!